### PR TITLE
CRD support

### PR DESCRIPTION
**Pull request by novakov-alexey-zz**
- Custom Resource Definition API
- `Filterable` trait for Label Selector functionality in the all APIs.

### CODE REVIEW

**Comment by novakov-alexey-zz** [comment]
@joan38 Do you have idea how to suppres deprecation warnings/errors on cross-compilation?

**Comment by joan38** [comment]
`mapValues` does not exists on Scala 2.12 collections, so your code is not cross compiling.
To fix that I imported [scala-collection-compat](https://github.com/scala/scala-collection-compat) and you can add this import:
```
import scala.collection.compat._
```

**Comment by novakov-alexey-zz** [comment]
it should close issue #5 as well

**Comment by novakov-alexey-zz** [comment]
@joan38 could you code-review this PR?

**Comment by joan38** [comment]
I'm on it @novakov-alexey, will circle back here soon. Thanks

File: src/main/scala/com/goyeau/kubernetes/client/api/Filterable.scala [hunk 1]
```diff
@@ -1,6 +1,6 @@
 package com.goyeau.kubernetes.client.api
 
-trait LabelSelector[A] {
+trait Filterable[A] {
```
Comment (joan38) [inline src/main/scala/com/goyeau/kubernetes/client/api/Filterable.scala]: I'm not a big fan of adding data on the resource objects that will then affect the actions (list, delete...) we do on it.
So I went ahead and pushed a commit on top of your branch to move that at the function level.
Let me know what you think and if I'm missing something.

Comment (novakov-alexey-zz) [reply src/main/scala/com/goyeau/kubernetes/client/api/Filterable.scala]: Ok, I am fine with that. There can be more such filters for bulk operations for example fieldSelector. We would need to add more parameters at function level later

**Pull request merged**